test/integration/compile-test.coffee

```diff
@@ -296,218 +306,311 @@ describe('compile() · all API description formats', ->
+      )
+      it('is compiled with no errors', ->
+        assert.deepEqual(compilationResult.errors, [])
+      )
       it('expands the request URI using correct inheritance cascade', ->
-        assert.equal(transaction.request.uri, '/honey?beekeeper=Honza&amount=42')
+        assert.equal(compilationResult.transactions[0].request.uri, '/honey?beekeeper=Honza&amount=42')
       )
     )
   )
 
   describe('with different default value and first enum value of URI parameter', ->
-    transaction = undefined
-
     fixtures.preferDefault.forEachDescribe(({source}) ->
+      compilationResult = undefined
+
       beforeEach((done) ->
         compileFixture(source, (args...) ->
           [err, compilationResult] = args
-          transaction = compilationResult.transactions[0]
           done(err)
         )
       )
 
+      it('is compiled with no warnings', ->
+        assert.deepEqual(compilationResult.warnings, [])
+      )
+      it.skip('is compiled with no errors', ->
+        assert.deepEqual(compilationResult.errors, [])
+      )
       it('expands the request URI using the default value', ->
-        assert.equal(transaction.request.uri, '/honey?beekeeper=Adam')
+        assert.equal(compilationResult.transactions[0].request.uri, '/honey?beekeeper=Adam')
       )
     )
   )
 
   describe('with default value for a required URI parameter', ->
-    errors = undefined
-    warnings = undefined
-    warning = undefined
-    transactions = undefined
-
     fixtures.defaultRequired.forEachDescribe(({source}) ->
+      compilationResult = undefined
+
       beforeEach((done) ->
         compileFixture(source, (args...) ->
           [err, compilationResult] = args
-          {errors, warnings, transactions} = compilationResult
-          [warning] = warnings[-1..] # the last warning
           done(err)
         )
       )
 
       it('expands the request URI using the default value', ->
-        assert.equal(transactions[0].request.uri, '/honey?beekeeper=Honza')
+        assert.equal(compilationResult.transactions[0].request.uri, '/honey?beekeeper=Honza')
       )
       it('is compiled with no errors', ->
-        assert.equal(errors.length, 0)
+        assert.deepEqual(compilationResult.errors, [])
       )
       it('is compiled with warnings', ->
-        assert.ok(warnings.length)
+        assert.ok(compilationResult.warnings.length)
       )
       it('there are no other warnings than from parser or URI expansion', ->
-        assert.equal(warnings.filter((w) ->
+        assert.equal(compilationResult.warnings.filter((w) ->
           w.component isnt 'uriTemplateExpansion' and
           w.component isnt 'apiDescriptionParser'
         ).length, 0)
       )
       context('the last warning', ->
         it('comes from URI expansion', ->
-          assert.equal(warning.component, 'uriTemplateExpansion')
+          assert.equal(compilationResult.warnings[-1..][0].component, 'uriTemplateExpansion')
```

Comment:
I have no idea what `array[-1..]` magic does. Is there a way to make this more explicit?